### PR TITLE
chore(integration): print test results after integration test finished

### PR DESCRIPTION
**Pull request by chuijiaolianying**
After all integration test cases finished, it would be better have a brief report to check test results.

The test results in this PR show like this:

```
TEST                                               | STATUS     | DURATION
HandlingDescendantsOfProposed                      | Passed     | 5 s
RelayInvalidTransaction                            | Passed     | 5 s
AlertPropagation                                   | Passed     | 17 s
ChainFork4                                         | Passed     | 19 s
...
BlockSyncWithUncle                                 | Failed     | 9 s
Total elapsed time: 425.92992889s
```

### CODE REVIEW

File: test/src/main.rs [hunk 1]
```diff
@@ -20,6 +20,13 @@ use std::sync::atomic::Ordering;
 use std::sync::Arc;
 use std::time::{Duration, Instant};
 
+struct TestResult {
+    spec_name: String,
+    status_name: String,
```
Comment (quake) [inline test/src/main.rs:25]: a minor suggestion: create a enum for TestResult status,
```
enum TestResultStatus {
  Pass,
  Panic,
  Fail
}
```

and implement `Ord` for this enum, then we can remove the `status_weight` from `TestResult` and sort the results by this status field.

Comment (chuijiaolianying) [reply test/src/main.rs]: Done, please take another look, thx

**Comment by keroro520** [comment]
> After all integration test cases finished, it would be better to have a brief report to check test results.
> 
> The test results in this PR show like this:
> 
> ```
> TEST                                               | STATUS     | DURATION
> HandlingDescendantsOfProposed                      | Passed     | 5 s
> RelayInvalidTransaction                            | Passed     | 5 s
> AlertPropagation                                   | Passed     | 17 s
> ChainFork4                                         | Passed     | 19 s
> ...
> BlockSyncWithUncle                                 | Failed     | 9 s
> Total elapsed time: 425.92992889s
> ```

The current code implementation uses the naming "Pass/Fail/Panic". I think the naming in your description is better.

**Review by keroro520 — APPROVED** [review]

**Review by keroro520 — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Comment by quake** [comment]
bors r=quake,keroro520

**Comment by quake** [comment]
compilation error, please resolve it

**Review by quake — APPROVED** [review]

**Review by keroro520 — APPROVED** [review]

**Comment by keroro520** [comment]
bors r=quake,keroro520

**Comment by yangby-cryptape** [comment]
@chuijiaolianying

Could u add an argument in command line to control this feature (or use `log::debug!(..)` so I can disable it by log filter)?
Sentry only keep the last 100 lines as `breadcrumbs`. (u can open our sentry to see an example for any integration tests issues.)

**Pull request merged**